src/expr.cpp

```diff
@@ -3121,14 +3140,14 @@ static llvm::Value *lEmitOpAssign(AssignExpr::Op op, Expr *arg0, Expr *arg1, con
     case AssignExpr::ModAssign:
     case AssignExpr::AddAssign:
     case AssignExpr::SubAssign:
-        newValue = lEmitBinaryArith(basicop, oldLHS, rvalue, type, arg1->GetType(), ctx, pos);
+        newValue = lEmitBinaryArith(basicop, oldLHS, rvalue, type, arg1->GetType(), ctx, pos, isSigned);
         break;
     case AssignExpr::ShlAssign:
     case AssignExpr::ShrAssign:
     case AssignExpr::AndAssign:
     case AssignExpr::XorAssign:
     case AssignExpr::OrAssign:
-        newValue = lEmitBinaryBitOp(basicop, oldLHS, rvalue, arg0->GetType()->IsUnsignedType(), ctx);
+        newValue = lEmitBinaryBitOp(basicop, oldLHS, rvalue, !arg0->GetType()->IsUnsignedType(), ctx);
```

Comment:
I think we discussed it already, check for "not-unsigned" should not be used, but "is signed" should be used instead is the intention is to check if this is a signed value.

Reply:
This is "resolved" now, but it's very difficult to track the changes and understand what's changed, when in the big change set the new version which was "push forced" and there's no clear explanation of the essence of the changes.